src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/Http2Tests.cs

```diff
@@ -0,0 +1,1017 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http2Cat;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IntegrationTesting.Common;
+using Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests.InProcess
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class Http2Tests : IISFunctionalTestBase
```

Comment:
I'm worried about test time with all of these tests.

Reply:
These tests average 2.5 seconds each, for a total of 43 seconds.

These tests don't need anything special except https, can we create an https fixture they can share?

Reply:
You can, but no need to block on doing that to implement it. Adding fixtures here is kind of tricky.

Reply:
https://github.com/dotnet/aspnetcore/issues/25384